src/lib/logger/langchain/tracer.ts

```
@@ -0,0 +1,502 @@
+type HandleLLMErrorParameters = Parameters<NonNullable<BaseCallbackHandler["handleLLMError"]>>;
+type HandleChatModelStartParameters = Parameters<NonNullable<BaseCallbackHandler["handleChatModelStart"]>>;
+type HandleRetrieverStartParameters = Parameters<NonNullable<BaseCallbackHandler["handleRetrieverStart"]>>;
+type HandleRetrieverEndParameters = Parameters<NonNullable<BaseCallbackHandler["handleRetrieverEnd"]>>;
+type HandleChainStartParameters = Parameters<NonNullable<BaseCallbackHandler["handleChainStart"]>>;
+type HandleChainEndParameters = Parameters<NonNullable<BaseCallbackHandler["handleChainEnd"]>>;
+type HandleToolStartParameters = Parameters<NonNullable<BaseCallbackHandler["handleToolStart"]>>;
+type HandleToolEndParameters = Parameters<NonNullable<BaseCallbackHandler["handleToolEnd"]>>;
+type HandleToolErrorParameters = Parameters<NonNullable<BaseCallbackHandler["handleToolError"]>>;
+
+export class MaximLangchainTracer extends BaseCallbackHandler {
+	override name: string = "MaximLangchainTracer";
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)